ejb3/src/main/java/org/jboss/as/ejb3/remote/AssociationImpl.java

```diff
@@ -217,6 +230,34 @@ public CancelHandle receiveInvocationRequest(@NotNull final InvocationRequest in
         return cancellationFlag::cancel;
     }
 
+    private static <A> boolean intersect(final Set<A> fst, final Set<A> snd) {
+        fst.retainAll(snd);
+        return !fst.isEmpty();
+    }
+
+    private static Set<String> typesOf(final Object o) {
```

Comment:
@dmlloyd this is so we could exclude any number of types from being passed through the call even if these types are not direct supertypes of the object passed in. This is technically not necessary for the case when we want to exclude weld's SerializableContextualInstanceImpl but I wanted to have some protection from the unpleasant surprises due to counter-intuitive behavior of Class.getSuperInterfaces. 
I thought that might be an overkill, but on the other hand both sets are very small.
The alternative is to do smth like EXCLUDED_CLASSNAMES.contains(o.getClass().getCanonicalNames()), that would be the fastest. I was just trying to make it look less like a crutch.